androguard/core/bytecodes/apk.py

```diff
@@ -913,75 +975,68 @@ def getString(self, idx):
 
         offset = self.m_stringOffsets[idx]
 
-        if not self.m_isUTF8:
-            length = self.getShort2(self.m_strings, offset)
-            offset += 2
-            self._cache[idx] = self.decode(self.m_strings, offset, length)
+        if self.m_isUTF8:
+            self._cache[idx] = self.decode8(offset)
         else:
-            offset += self.getVarint(self.m_strings, offset)[1]
-            varint = self.getVarint(self.m_strings, offset)
-
-            offset += varint[1]
-            length = varint[0]
-
-            self._cache[idx] = self.decode2(self.m_strings, offset, length)
+            self._cache[idx] = self.decode16(offset)
 
         return self._cache[idx]
 
     def getStyle(self, idx):
-        print idx
-        print idx in self.m_styleOffsets, self.m_styleOffsets[idx]
+        # FIXME
```

Comment:
the problem was that I don't know how to check this function, and what it should have returned in the first place. Before it just had a number of printouts and returned nothing:) so FIXME here means "Please review this function someone who knows what it should do"